engine.py

```diff
@@ -11,8 +11,28 @@
 import sys
 
 import sgtk
+import traceback
 from sgtk.platform import Engine
 
+previous_except_hook = sys.excepthook
+
+
+def unhandled_exception_handler(exc_type, exc_value, exc_traceback):
+    # Calls any previous exception handler. By default, this will invoke the
+    # console error handler.
+    if previous_except_hook:
+        previous_except_hook(exc_type, exc_value, exc_traceback)
+
+    engine = sgtk.platform.current_engine()
+    # As long as we have an engine, we can send errors to its logger.
+    if engine:
+        engine.logger.error(
+            "".join(traceback.format_exception(exc_type, exc_value, exc_traceback))
+        )
```

Comment:
I'm not sure on when events where this method is called, but this will catch any errors raised after the engine has started. What if an app errors during initialisation, does it get logged then by some other code?

Reply:
Ah, good point. If the engine hasn't started yet, we're going to miss it. Maybe I should log directly to `sgtk.LogManager.get_logger(__file__)` instead! I'll try it out. Thanks!

Reply:
Nice!